server/plugins/IOUGens.cpp

```diff
@@ -1327,15 +1327,20 @@ void OffsetOut_Ctor(OffsetOut* unit) {
     unit->m_busTouched = world->mAudioBusTouched;
     int32 offset = unit->mParent->mSampleOffset;
     int numChannels = unit->mNumInputs - 1;
-
-    unit->m_saved = (float*)RTAlloc(unit->mWorld, offset * numChannels * sizeof(float));
-    ClearUnitIfMemFailed(unit->m_saved);
+    // NB: if mSampleOffset is 0, RTAlloc() might return a nullptr and
+    // trigger ClearUnitIfMemFailed(), so we have to handle it specially.
+    if (offset > 0) {
+        unit->m_saved = (float*)RTAlloc(unit->mWorld, offset * numChannels * sizeof(float));
+        ClearUnitIfMemFailed(unit->m_saved);
+    } else {
+        unit->m_saved = nullptr;
+    }
     unit->m_empty = true;
     // Print("<-Out_Ctor\n");
 }
 
 void OffsetOut_Dtor(OffsetOut* unit) {
-    // Ctor may not have run, if unit was immediately created->paused->freed!
```

Comment:
Was this comment inaccurate? With so few people understand what's going on in SN, I wonder if we should remove breadcrumbs like this.

Reply:
Yes, the comment is wrong/misleading because a unit destructor only gets called if the constructor has been called.

Scsynth:
https://github.com/supercollider/supercollider/blob/02c37d28badc390f7ee320883b955052294d3cd6/server/scsynth/SC_Graph.cpp#L56-L65

Supernova:
https://github.com/supercollider/supercollider/blob/02c37d28badc390f7ee320883b955052294d3cd6/server/supernova/sc/sc_synth.cpp#L171-L176